PWG/EMCAL/EMCALtasks/AliEmcalMCTrackSelector.h

```diff
@@ -115,58 +115,72 @@ class AliEmcalMCTrackSelector : public AliAnalysisTaskSE {
 
   /**
    * @brief Set the name of the output container
-   * 
-   * This container is attached to the input event with the corresponding name. 
+   *
+   * This container is attached to the input event with the corresponding name.
    * This name has to be used in the user tasks to connect the MC particle container
    * to the particles selected by this instance of the task.
-   * 
+   *
    * @param name Name of the output container attached to the input event
    */
   void SetParticlesOutName(const char *name)            { fParticlesOutName = name ; }
 
+  /**
+   * @brief Set the name of the input track container
+   *
+   * If there is a track container attached to the event other than the default,
+   * this is where to specify the name. For example, if AliEmcalRejectMCBackground
+   * is run first, it will attach a new track container to the event. Otherwise,
+   * leave this as default.
+   *
+   * @param nTrackCont Name of the track container attached to the input event
+   */
+  void SetTrackContainerName(const char *nTrackCont)    {fTrackContainerName = nTrackCont ; }
+
+  void SetIsContainerSpecified(bool isContSpecified)    {fIsContainerSpecified = isContSpecified ; }
+
 
   /**
    * @brief Create new AliEmcalMCTrackSelector task and add it to the analysis manager
-   * 
+   *
    * @param outname name of the output contaienr
    * @param nk Reject neutrons and K0long
    * @param ch Select only charged particles
    * @param etamax  Max eta acceptance
    * @param physPrim Require physical primary particles
-   * @return AliEmcalMCTrackSelector* 
+   * @return AliEmcalMCTrackSelector*
    */
-  static AliEmcalMCTrackSelector* AddTaskMCTrackSelector(TString outname = "mcparticles", Bool_t nk = kFALSE, Bool_t ch = kFALSE, Double_t etamax = 1, Bool_t physPrim = kTRUE);
+  static AliEmcalMCTrackSelector* AddTaskMCTrackSelector(TString outname = "mcparticles", TString nTrackCont = "default", Bool_t nk = kFALSE, Bool_t ch = kFALSE, Double_t etamax = 1, Bool_t physPrim = kTRUE);
```

Comment:
Please use `usedefault` as default argument for incoming containers in order to align with other tasks.